tests/basic/sharedv4_svc_test.go

```diff
@@ -199,648 +196,603 @@ type appCounter struct {
 	active bool
 }
 
-// Induce failovers and verify that there is no I/O disruption.
-var _ = Describe("{Shared4SvcFailoverIO}", func() {
-	var testrailID = 54374
-	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/54374
-	var runID int
+type failoverMethod interface {
+	doFailover(attachedNode *node.Node)
+	getExpectedPodDeletions() []int
+	String() string
+}
+
+type failoverMethodRestartVolDriver struct {
+}
+
+func (fm *failoverMethodRestartVolDriver) doFailover(attachedNode *node.Node) {
+	restartVolumeDriverOnNode(attachedNode)
+}
+
+func (fm *failoverMethodRestartVolDriver) String() string {
+	return "restart volume driver"
+}
+
+func (fm *failoverMethodRestartVolDriver) getExpectedPodDeletions() []int {
+	// 2 pods, one on the old NFS server and one on the new NFS server should be deleted.
+	return []int{2}
+}
+
+type failoverMethodReboot struct {
+}
+
+func (fm *failoverMethodReboot) doFailover(attachedNode *node.Node) {
+	rebootNodeAndWaitForReady(attachedNode)
+}
+
+func (fm *failoverMethodReboot) String() string {
+	return "reboot"
+}
+
+func (fm *failoverMethodReboot) getExpectedPodDeletions() []int {
+	// 1 or 2 pods may get deleted depending on what kubelet does on the rebooted node.
+	// The kubelet could set up the mount for the same pod or it could create a new pod.
+	return []int{1, 2}
+}
+
+var _ = Describe("{Shared4 service apps}", func() {
+	var testrailID, runID int
 	var contexts, testSv4Contexts []*scheduler.Context
+	var workers []node.Node
 	var numPods int
+	var namespacePrefix string
 
 	JustBeforeEach(func() {
 		runID = testrailuttils.AddRunsToMilestone(testrailID)
-	})
 
-	It("has to verify no I/O disruption for test-sv4-svc apps after PX restart or reboot on NFS server node", func() {
-		contexts = make([]*scheduler.Context, 0)
+		// Set up all apps
+		contexts = nil
```

Comment:
Is setting contexts to nil necessary here? I don't see it being populated before this test step. Since it's nil by default when it's defined, i don't see a need to set it to nil again.